builtin/game/privileges.lua

```diff
@@ -8,13 +8,13 @@ core.registered_privileges = {}
 
 function core.register_privilege(name, param)
 	local function fill_defaults(def)
-		if def.give_to_singleplayer == nil then
+		if not def.give_to_singleplayer then
 			def.give_to_singleplayer = true
 		end
-		if def.give_to_admin == nil then
+		if not def.give_to_admin then
 			def.give_to_admin = def.give_to_singleplayer
 		end
```

Comment:
The comparisons with `nil` are needed here as `false` has to stay possible.
Something you could however do is eg. `def.give_to_singleplayer = def.give_to_singleplayer ~= false`.

Reply:
Thanks, will fix it later

Reply:
OLD:
```
if def.give_to_singleplayer == nil then
	def.give_to_singleplayer = true
end
if def.give_to_admin == nil then
	def.give_to_admin = def.give_to_singleplayer
end
```
NEW:
```
def.give_to_singleplayer = def.give_to_singleplayer ~= false
def.give_to_admin = def.give_to_admin ~= false and def.give_to_singleplayer
```

Reply:
(Oh, I didn't see that it's `def.give_to_admin = def.give_to_singleplayer` and not `def.give_to_admin = true`.)

`def.give_to_admin = def.give_to_admin ~= false and def.give_to_singleplayer` isn't correct, it should rather be `def.give_to_admin = def.give_to_admin ~= false and def.give_to_singleplayer or def.give_to_admin`.

It's probably best to keep the if with comparison with `nil`.

Reply:
I reverted this change. Thanks for the help, did not think about it. 👍

I would be grateful if you check other changes.